compiler/optimizer/InductionVariable.cpp

```diff
@@ -4223,9 +4223,30 @@ void TR_LoopStrider::detectLoopsForIndVarConversion(
    ListIterator<TR::Block> bi (&blocksInRegion);
    TR::Block *nextBlock;
    for (nextBlock = bi.getCurrent(); nextBlock; nextBlock = bi.getNext())
+      {
       if (nextBlock->hasExceptionPredecessors())
          return;
 
+      //in fsd, bail out if there is any exception edges to OSRCatchBlock because there is no
+      //safe way to split those edges. Also, there is no need to check if the OSRCatchBlock is 
+      //within the loop because loop strider already skips any loop that contains a 
+      //catch block
+      if (TR::comp()->getOption(TR_FullSpeedDebug) && nextBlock->hasExceptionSuccessors())
```

Comment:
Is there any reason why FSD must be checked here ? 

It might be more general to skip this if OSR catch blocks and exception edges end up used in other cases.